ansibleinviewer/inventoryadapter.py

```diff
@@ -32,3 +32,26 @@ def get_hosts_by_names(self, hostnames: List[str]) -> List[Host]:
         output_hosts = {self._inventory.hosts[hostname] for hostname in hostnames if
                         hostname in self._inventory.hosts}
         return list(output_hosts)
+
+    def get_hosts_by_variables(self, hosts: List,
+                               variables: List, no_variables: List) -> List[Host]:
+        output_hosts = []
+        if not no_variables:
+            no_variables = []
+        variable_husked_hosts = []
+        if not variables:
+            variable_husked_hosts = hosts
+        else:
+            for host in hosts:
+                if any(var in host.get_vars().items()
+                       for var in variables) or \
+                   any(var[0] in host.get_vars().keys()
+                       for var in variables if len(var) < 2):
```

Comment:
I think it would be more readable to have a list of tuples, where there is (key, val) or (key, None) if there is no val provided. Then the obscure `if len(var) < 2` could be changed to `if var[1]`.